code/build/template/include/extensions/[+python orxPy.h]

```diff
@@ -0,0 +1,2613 @@
+    orxVECTOR vRes = orxVECTOR_0;           \
+    if (py_isfloat(pyOp) || py_isint(pyOp)) \
+    {                                       \
+      ARG_FLOAT(fOp, 1);                    \
+      func##f(&vRes, vSelf, fOp);           \
+    }                                       \
+    else                                    \
+    {                                       \
+      ARG_VECTOR(vOp, 1);                   \
+      func(&vRes, vSelf, vOp);              \
+    }                                       \
+    RETURN_VECTOR(&vRes);                   \
+  }
+
+  BIND_VECTOR_OP(__add__, orxVector_Add)
+  BIND_VECTOR_OP(__sub__, orxVector_Sub)
+  BIND_VECTOR_MOP(__mul__, orxVector_Mul)
+  BIND_VECTOR_MOP(__truediv__, orxVector_Div)
+
+#undef BIND_VECTOR_OP
+#undef BIND_VECTOR_MOP
+
+#define BIND_VECTOR_PROPERTY(property) \
+  BIND(Vector, get_##property)         \
+  {                                    \
+    PY_CHECK_ARGC(1);                  \
+    ARG_VECTOR(pvSelf, 0);             \
+    RETURN_FLOAT(pvSelf->f##property); \
+  }                                    \
+  BIND(Vector, set_##property)         \
+  {                                    \
+    PY_CHECK_ARGC(2);                  \
+    ARG_VECTOR(pvSelf, 0);             \
+    ARG_FLOAT(f##property, 1);         \
+    pvSelf->f##property = f##property; \
+    RETURN_NONE;                       \
+  }
+
+  BIND_VECTOR_PROPERTY(X)
+  BIND_VECTOR_PROPERTY(Y)
+  BIND_VECTOR_PROPERTY(Z)
+
+#undef BIND_VECTOR_PROPERTY
+
+  BIND(Vector, size)
+  {
+    PY_CHECK_ARGC(1);
+    ARG_VECTOR(vOp, 0);
+    RETURN_FLOAT(orxVector_GetSize(vOp));
+  }
+
+  BIND(Vector, distance)
+  {
+    PY_CHECK_ARGC(2);
+    ARG_VECTOR(vOp1, 0);
+    ARG_VECTOR(vOp2, 1);
+    RETURN_FLOAT(orxVector_GetDistance(vOp1, vOp2));
+  }
+
+  BIND(Vector, lerp)
+  {
+    PY_CHECK_ARGC(3);
+    ARG_VECTOR(vOp1, 0);
+    ARG_VECTOR(vOp2, 1);
+    ARG_FLOAT(fOp, 2);
+    orxVECTOR vRes = orxVECTOR_0;
+    orxVector_Lerp(&vRes, vOp1, vOp2, fOp);
+    RETURN_VECTOR(&vRes);
+  }
+
+  BIND(Vector, rotate_in_place)
+  {
+    PY_CHECK_ARGC(2);
+    ARG_VECTOR(vSelf, 0);
+    ARG_FLOAT(fAngle, 1);
+    orxVector_2DRotate(vSelf, vSelf, fAngle);
+    RETURN_NONE;
+  }
+
+  BIND(Vector, rotate)
+  {
+    PY_CHECK_ARGC(2);
+    ARG_VECTOR(vOp, 0);
+    ARG_FLOAT(fAngle, 1);
+    orxVECTOR vRes = orxVECTOR_0;
+    orxVector_2DRotate(&vRes, vOp, fAngle);
+    RETURN_VECTOR(&vRes);
+  }
+
+  BIND(Vector, normalize_in_place)
+  {
+    PY_CHECK_ARGC(1);
+    ARG_VECTOR(vSelf, 0);
+    orxVector_Normalize(vSelf, vSelf);
+    RETURN_NONE;
+  }
+
+  BIND(Vector, normalize)
+  {
+    PY_CHECK_ARGC(1);
+    ARG_VECTOR(vOp, 0);
+    orxVECTOR vRes = orxVECTOR_0;
+    orxVector_Normalize(&vRes, vOp);
+    RETURN_VECTOR(&vRes);
+  }
+
+  BIND(Clock, compute_dt)
+  {
+    PY_CHECK_ARGC(2);
+    ARG_CLOCK(pstClock, 0);
+    ARG_FLOAT(fDT, 1);
+    RETURN_FLOAT(orxClock_ComputeDT(pstClock, fDT));
+  }
+
+  void orxPy_AddModuleClock()

```

Comment:
The `orxPy_AddModule*` functions define the Python modules which correspond to each bound `orx*` module. They map C definitions to Python definitions.